pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -268,15 +272,16 @@ var (
 )
 
 // NewKV creates new gossip-based KV service. Note that service needs to be started, until then it doesn't initialize
-// gossiping part. Only after service is in Running state, it is really gossiping.
-// Starting the service will also trigger connecting to the existing memberlist cluster, if cfg.JoinMembers is set.
-// If that fails and AbortIfJoinFails is true, error is returned and service enters Failed state.
+// gossiping part. Only after service is in Running state, it is really gossiping. Starting the service will also
+// trigger connecting to the existing memberlist cluster. If that fails and AbortIfJoinFails is true, error is returned
+// and service enters Failed state.
 func NewKV(cfg KVConfig) *KV {
 	cfg.TCPTransport.MetricsRegisterer = cfg.MetricsRegisterer
 	cfg.TCPTransport.MetricsNamespace = cfg.MetricsNamespace
 
 	mlkv := &KV{
 		cfg:            cfg,
+		provider:       dns.NewProvider(util.Logger, cfg.MetricsRegisterer, dns.GolangResolverType),
```

Comment:
The Prometheus registerer passed to the DNS provider should be something like this (to keep it consistent with other components and avoid "duplicated metrics registration"):

```
dnsProviderRegisterer := prometheus.WrapRegistererWithPrefix("cortex_", prometheus.WrapRegistererWith(prometheus.Labels{
		"name": "memberlist",
	}, cfg.MetricsRegisterer))
```

@pstibrany Make sense to you? Is it safe to use an hardcoded `memberlist` name?

Reply:
IC, the `cfg.MetricsRegisterer` is only the defaultRegisterer up in the stack `pkg/cortex/modules.go` for memberlist. It makes sense to me to separate the provider metrics per component.